File: modules/spring-mock-mvc/src/test/java/io/restassured/module/mockmvc/http/GreetingController.java
/*
 * Copyright 2019 the original author or authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package io.restassured.module.mockmvc.http;

import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.ResponseBody;

import java.util.concurrent.atomic.AtomicLong;

import static org.springframework.web.bind.annotation.RequestMethod.GET;
import static org.springframework.web.bind.annotation.RequestMethod.POST;

/**
 * This controller has been copied from <a href="http://spring.io/guides/gs/rest-service/">Spring Guides</a>.
 */
@Controller
public class GreetingController {

    private static final String template = "Hello, %s!";
    private final AtomicLong counter = new AtomicLong();

    @RequestMapping(value = "/greeting", method = GET)
    public @ResponseBody Greeting greeting(
            @RequestParam(value="name", required=false, defaultValue="World") String name) {
        return new Greeting(counter.incrementAndGet(), String.format(template, name));
    }

    @RequestMapping(value = "/greeting", method = POST, consumes = "application/json", produces = "application/json")
    public @ResponseBody Greeting greetingWithRequiredContentType(
            @RequestParam(value="name", required=false, defaultValue="World") String name) {
        return greeting(name);
    }
}

File: modules/spring-mock-mvc-kotlin-extensions/src/test/kotlin/io/restassured/module/mockmvc/kotlin/extensions/GreetingController.kt
package io.restassured.module.mockmvc.kotlin.extensions

import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestMethod
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.ResponseBody
import java.util.concurrent.atomic.AtomicLong

@Controller
class GreetingController {
    private val template = "Hello, %s!"
    private val counter = AtomicLong()

    @RequestMapping(
        value = ["/greeting"],
        method = [RequestMethod.GET]
    )
    @ResponseBody
    fun greeting(
        @RequestParam(
            value = "name",
            required = false,
            defaultValue = "World"
        ) name: String
    ) = Greeting(
        counter.incrementAndGet(),
        String.format(template, name)
    )

    @RequestMapping(
        value = ["/greeting"],
        method = [RequestMethod.POST],
        consumes = ["application/json"],
        produces = ["application/json"]
    )
    @ResponseBody
    fun greetingWithRequiredContentType(
        @RequestParam(
            value = "name",
            required = false,
            defaultValue = "World"
        ) name: String
    ) = greeting(name)
}

data class Greeting(
    val id: Long,
    val content: String
)
